Add paimon recipe

r0man opened the pull request:
### Brief summary of what the package does

An Emacs mode for [Splunk](https://splunk.org/).

### Direct link to the package repository

https://github.com/r0man/paimon.el

### Your association with the package

I'm the maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks for this!  Very quick first pass --

`paimon-profile.el` with byte-compile using Emacs 27.1:
```
paimon-profile.el:134:8:Warning: read-number called with 3 arguments, but
    accepts only 1-2
```

`paimon-profiles.el` with [melpazoid](https://github.com/riscy/melpazoid):
- paimon-profiles.el#L128: This is unnecessary under `define-derived-mode` -- same with setting the mode-name, setting the keymap, and running the hooks -- you may in fact be causing hooks to run twice

`paimon-search-jobs.el` with [melpazoid](https://github.com/riscy/melpazoid):
- paimon-search-jobs.el#L122: This double negation can be collapsed (`not` aliases `null`)
- paimon-search-jobs.el#L252: Same as the comment on `paimon-profiles.el`

`paimon-search-result.el` with [melpazoid](https://github.com/riscy/melpazoid):
- paimon-search-result.el#L133: Same as previous comments

`paimon-search-results.el` with [melpazoid](https://github.com/riscy/melpazoid):
- paimon-search-results.el#L167: It's safer to sharp-quote function names; use `#'`
- paimon-search-results.el#L162: Same as previous comments

`paimon-search.el` with package-lint 20211122.1152:
```
5 issues found:
78:0: error: `paimon-search:--earliest-time' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
85:0: error: `paimon-search:--indexes' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
93:0: error: `paimon-search:--latest-time' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
100:0: error: `paimon-search:--search-level' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
107:0: error: `paimon-search:--status-buckets' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
```

`paimon-util.el` with [melpazoid](https://github.com/riscy/melpazoid):
- paimon-util.el#L104: No `format` required; user-error takes an f-string

Package and license:
- Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
  See: https://github.com/licensee/licensee
- Only a couple of the files have identifiable license boilerplate -- the ones that don't are below
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-api.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-data-index.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-profile.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-profiles.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-search-job.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-search-result.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-search-results-layout.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-search-results.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-search.el
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to paimon-util.el

r0man commented:
Hi riscy! 

Thanks for the review, I learned quite a bit! 
I fixed the issues you mentioned, and also added your build action to the project and it passes. 

Roman

riscy commented:
Great, thank you for this!
